pkg/services/sso/keycloak_service_builder.go

```diff
@@ -0,0 +1,55 @@
+package sso
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/keycloak"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/redhatsso"
+)
+
+var _ KeycloakServiceBuilderSelector = &keycloakServiceBuilderSelector{}
+var _ KeycloakServiceBuilder = &keycloakServiceBuilder{}
+
+type KeycloakServiceBuilder interface {
+	WithRealmConfig(realmConfig *keycloak.KeycloakRealmConfig) KeycloakServiceBuilder
+	Build() KeycloakService
+}
+
+type keycloakServiceBuilder struct {
+	config      *keycloak.KeycloakConfig
+	realmConfig *keycloak.KeycloakRealmConfig
+}
+
+func (builder *keycloakServiceBuilder) Build() KeycloakService {
+	if builder.config.SelectSSOProvider == keycloak.MAS_SSO {
+		realmConfig := builder.realmConfig
+		if realmConfig == nil {
+			realmConfig = builder.config.KafkaRealm
+		}
+		return newKeycloakService(builder.config, realmConfig)
+	} else {
+		client := redhatsso.NewSSOClient(builder.config)
+		return &keycloakServiceProxy{
+			accessTokenProvider: client,
```

Comment:
``` realmConfig  = builder.config.RedhatSSORealm ```
do we need to define it here as well?

Reply:
Currently redhatsso supports only one realconfig from the KetCloakConfig object.
If we need the ability to support a custom realm in redhatsso as we did in mas-sso, then I can make the change (it will be trivial)

Reply:
make sense

Reply:
It doesn't. Currently the RedhatSSO takes the realm directly from the config

Reply:
See https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/3773316adce8f9d17b62fcf90b97a72eaf59b716/pkg/client/redhatsso/client.go#L141